src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/__tests__/TargetingRulesTab.test.tsx

```diff
@@ -48,309 +48,491 @@ describe('TargetingRulesTab', () => {
+      userEvent.click(screen.getByTestId('remove_variation_false'))
 
-    expect(screen.queryByTestId('false_target_groups')).not.toBeInTheDocument()
-    expect(screen.queryByTestId('false_targets')).not.toBeInTheDocument()
+      expect(screen.queryByTestId('false_target_groups')).not.toBeInTheDocument()
+      expect(screen.queryByTestId('false_targets')).not.toBeInTheDocument()
+    })
   })
 
-  test('it should call endpoint with correct data on save', async () => {
-    const saveChangesMock = jest.fn()
+  describe('Percentage Rollout', () => {
+    test('it should render percentage rollout correctly when present', async () => {
+      renderComponent()
+
+      expect(screen.getByText('cf.featureFlags.percentageRollout')).toBeInTheDocument()
+      const targetGroup = document.querySelector('input[name="variationPercentageRollouts[0].clauses[0].values[0]"]')
+      expect(targetGroup).toHaveValue('target_group_1')
+      const trueWeight = document.querySelector('input[name="variationPercentageRollouts[0].variations[0].weight"]')
+      expect(trueWeight).toHaveValue(45)
+      const falseWeight = document.querySelector('input[name="variationPercentageRollouts[0].variations[1].weight"]')
+      expect(falseWeight).toHaveValue(55)
+    })
 
-    jest.spyOn(usePatchFeatureFlagMock, 'default').mockReturnValue({ saveChanges: saveChangesMock, loading: false })
-    renderComponent()
+    test('it should render percentage rollout correctly when added and not already present', async () => {
+      renderComponent({
+        featureFlagData: {
+          ...mockFeature,
+          envProperties: {
+            defaultServe: { variation: 'false' },
+            environment: 'qatest',
+            modifiedAt: 1635333973373,
+            offVariation: 'false',
+            rules: [
+              {
+                clauses: [
+                  {
+                    attribute: '',
+                    id: 'd36b6624-c514-4b94-94c7-9f558324badf',
+                    negate: false,
+                    op: 'segmentMatch',
+                    values: ['randomID']
+                  }
+                ],
+                priority: 100,
+                ruleId: '9dec5abb-002e-45b3-b241-963ac5d9acde',
+                serve: { variation: 'false' }
+              }
+            ],
+            state: 'off',
+            variationMap: [],
+            version: 56
+          }
+        }
+      })
+      expect(screen.queryByText('cf.featureFlags.percentageRollout')).not.toBeInTheDocument()
 
-    // toggle flag off
-    const flagToggle = screen.getByTestId('flag-status-switch')
-    userEvent.click(flagToggle)
-    expect(flagToggle).not.toBeChecked()
+      userEvent.click(screen.getByText('cf.featureFlags.rules.addTargeting'))
+      const percentageRolloutOption = screen.getByTestId('variation_option_percentage_rollout')
+      await waitFor(() => expect(percentageRolloutOption).toBeInTheDocument())
```

Comment:
This will either succeed or throw an error before the `waitFor` is reached. You might want to try `findBy` here instead of `getBy`
```suggestion

      const percentageRolloutOption = await screen.findByTestId('variation_option_percentage_rollout')

```